hw/ip/otbn/dv/otbnsim/sim/state.py

```diff
@@ -170,7 +209,17 @@ def start(self) -> None:
 
         # Reset CSRs, WSRs, loop stack and call stack
         self.csrs = CSRFile()
+
+        # Save the RND value when starting another run because when a SW
+        # run finishes we still keep RND data.
+        if self.wsrs.RND._random_value is not None:
+            self.rnd_cached_tmp = self.wsrs.RND._random_value
+
         self.wsrs = WSRFile()
+        if self.rnd_cached_tmp is not None:
+            self.wsrs.RND.set_unsigned(self.rnd_cached_tmp)
+            self.rnd_cached_tmp = None
```

Comment:
I don't think we need `rnd_cached_tmp` to be a field on the object. Can't we just do something like this?
```suggestion
        # Save the RND value when starting another run because when a SW
        # run finishes we still keep RND data.
        old_rnd = self.wsrs.RND._random_value

        self.wsrs = WSRFile()
        if old_rnd is not None:
            self.wsrs.RND.set_unsigned(old_rnd)
```

Reply:
oh it's much better this way, sorry!